In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('Documents/FoodBalanceSheets_E_Africa_NOFLAG.csv')

In [14]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [68]:
#What is the total sum of Animal Fat produced in 2014 and 2017 respectively?
print(df.groupby('Item')['Y2014'].sum())
print(" \n",df.groupby('Item')['Y2017'].sum())

Item
Alcohol, Non-Food        2403.00
Alcoholic Beverages    102410.11
Animal Products         11935.65
Animal fats            209460.54
Apples and products      9499.23
                         ...    
Vegetables, Other      155038.96
Vegetal Products       107145.19
Wheat and products     232670.13
Wine                     4497.36
Yams                   200396.96
Name: Y2014, Length: 119, dtype: float64
 
 Item
Alcohol, Non-Food        2348.00
Alcoholic Beverages     95581.06
Animal Products         11547.65
Animal fats            269617.53
Apples and products     10198.90
                         ...    
Vegetables, Other      157752.59
Vegetal Products       107655.20
Wheat and products     240047.62
Wine                     4178.02
Yams                   229174.59
Name: Y2017, Length: 119, dtype: float64


In [114]:
#What is the total sum of Animal Fat produced in 2014 and 2017 respectively?
print(round(df['Y2015'].mean(), 3))
print(round(df['Y2015'].std(), 3))

135.236
1603.404


In [115]:
#What is the total number and percentage of missing data in 2016 to 2 decimal places?
print("Number of missing data - ", df['Y2016'].isnull().sum())
print("Percentage of missing data - ", round(df['Y2016'].isnull().sum()*100 / len(df),2))

Number of missing data -  1535
Percentage of missing data -  2.52


In [116]:
#Which year had the highest correlation with ‘Element Code’? 
df.corr().loc['Element Code'].sort_values()

Item Code      -0.024683
Area Code      -0.000209
Y2016           0.023444
Y2015           0.023889
Y2017           0.024254
Y2018           0.024279
Y2014           0.024457
Element Code    1.000000
Name: Element Code, dtype: float64

In [106]:
print("=====================++++2014+++++=====================")
print(df.groupby('Element')['Y2014'].sum())
print("=====================++++2015+++++=====================")
print(df.groupby('Element')['Y2015'].sum())
print("=====================++++2016+++++=====================")
print(df.groupby('Element')['Y2016'].sum())
print("=====================++++2017+++++=====================")
print(df.groupby('Element')['Y2017'].sum())
print("=====================++++2018+++++=====================")
print(df.groupby('Element')['Y2018'].sum())

=====================++++2014+++++=====================
Element
Domestic supply quantity                  1996716.35
Export Quantity                            150020.64
Fat supply quantity (g/capita/day)          10225.56
Feed                                       216927.89
Food                                      1212332.49
Food supply (kcal/capita/day)              454257.00
Food supply quantity (kg/capita/yr)         49650.63
Import Quantity                            274144.48
Losses                                     153223.00
Other uses (non-food)                       78718.13
Processing                                 282923.00
Production                                1931287.75
Protein supply quantity (g/capita/day)      11836.46
Residuals                                   30149.00
Seed                                        21922.92
Stock Variation                             58749.83
Total Population - Both sexes             1031585.00
Tourist consumption                

#### From the results above, the questions below can be answered
##### What year has the highest sum of Import Quantity?
2017

##### What is the total number of the sum of Production in 2014?
1931287.75

##### Which of these elements had the highest sum in 2018?
Domestic supply quantity

##### Which of these elements had the 3rd lowest sum in 2018?
Protein supply quantity (g/capita/day)

In [117]:
#What is the total number of unique countries in the dataset?
len(df['Area'].value_counts())

49